### PR TITLE
Icicle now builds cleanly on GHC 8.0

### PR DESCRIPTION
**Pull request by jystic**
With the new jetski fixing the libffi problem, icicle can now be upgraded to GHC 8.0

! @amosr @tranma

### CODE REVIEW

File: icicle-compiler/src/Icicle/Avalanche/Statement/Simp/Constructor.hs [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+-- Disable pattern checks on OS/X because goX' below hits a pathalogical case
+-- in GHC and we don't want to wait for it on our dev machines.
+{-# OPTIONS_GHC -fno-warn-incomplete-patterns #-}
+{-# OPTIONS_GHC -fno-warn-incomplete-uni-patterns #-}
+{-# OPTIONS_GHC -fno-warn-overlapping-patterns #-}
+#else
+-- On the build bot we want to make sure the pattern matches are still safe but
+-- we need to increase a few limits to allow GHC to do this.
+{-# OPTIONS_GHC -fmax-pmcheck-iterations=10000000 #-}
+#endif
```
Comment (tranma) [inline icicle-compiler/src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: I missed the boat on this, what's the story?

Comment (jystic) [inline icicle-compiler/src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: There was a lot of lines changed so I just wanted to call this out

**Comment by tranma** [comment]
nice :1st_place_medal:

File: icicle-compiler/icicle.lock-7.10.2 [hunk 2]
```diff
@@ -10,11 +10,12 @@ attoparsec == 0.13.1.0
 base-orphans == 0.5.4
 base-unicode-symbols == 0.2.2.4
 bifunctors == 5.3
+bindings-DSL == 1.0.23
 blaze-builder == 0.4.0.2
 Boolean == 0.2.3
+Cabal == 1.24.1.0
```
Comment (amosr) [inline icicle-compiler/icicle.lock-7.10.2]: this is weird but talked IRL and not an issue

**Comment by amosr** [comment]
yep 👍

**Pull request merged**